services/wallet/router/router_updates.go

```diff
@@ -0,0 +1,127 @@
+package router
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/rpc/chain"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+)
+
+var (
+	newBlockCheckIntervalMainnet  = 2 * time.Millisecond
```

Comment:
every 2 milliseconds?

Reply:
I'd suggest every 6 seconds for mainnet. we will be a bit off but far from wrong/

Also what happens if in between u don't have enought to pay?

Reply:
Timeout set to 5 minutes, we're checking every 3 seconds for a new block on mainnet, if needed we will adjust that later (it's a single line change).

Reply:
For swaps, for example, we need to recalculate fees every 15 seconds https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=124-19384&t=UMFqL7n7Hgp00nTo-4